src/clients/bridges/ArbitrumAdapter.ts

```diff
@@ -130,7 +128,11 @@ export class ArbitrumAdapter extends BaseAdapter {
       // The logic below takes the results from the promises and spreads them into the l1DepositInitiatedEvents and
       // l2DepositFinalizedEvents state from the BaseAdapter.
       eventsToProcess.forEach((result, index) => {
-        const l1Token = validTokens[Math.floor(index / 2)];
+        if (eventsToProcess.length === 0) {
+          return;
+        }
+        assert(eventsToProcess.length % 2 === 0, "Events list length should be even");
+        const l1Token = availableL1Tokens[Math.floor(index / 2)];
```

Comment:
We can remove the Floor check because our length is even
```suggestion
        const l1Token = availableL1Tokens[index / 2];
```